ktor-server/ktor-server-core/jvm/src/io/ktor/routing/RouteSelector.kt

```diff
@@ -201,15 +217,43 @@ public data class OptionalParameterRouteSelector(val name: String) :
  * Evaluates a route against a constant path segment
  * @param value is a value of the path segment
  */
-public data class PathSegmentConstantRouteSelector(val value: String) :
-    RouteSelector(RouteSelectorEvaluation.qualityConstant) {
-    override fun evaluate(context: RoutingResolveContext, segmentIndex: Int): RouteSelectorEvaluation {
-        if (segmentIndex < context.segments.size && context.segments[segmentIndex] == value)
-            return RouteSelectorEvaluation.ConstantPath
-        return RouteSelectorEvaluation.Failed
+public data class PathSegmentConstantRouteSelector(val value: String) : RouteSelector(RouteSelectorEvaluation.qualityConstant) {
+
+    private var hasTrailingSlash: Boolean = false
+
+    public constructor(value: String, hasTrailingSlash: Boolean): this(value) {
```

Comment:
So what if i create a segment selector for something in the middle, not last component?
 And why we don't expose the property? Shouldn't equals consider it and why?

Reply:
Sorry, this branch was outdated and needed a rebase. This change is obsolete now.